src/v1_0/rw/elog.cpp

```diff
@@ -0,0 +1,140 @@
+    {
+        if (li_element.getAttribute("class") != "elog-message-ev")
+            return;
+
+        std::string uri;
+
+        auto nodes = li_element.childNodes();
+        for (int i = 0; i < nodes->length(); i++)
+        {
+            auto node = nodes->item(i);
+            if (xmlNodeHasAttribute(node, "rel", "self"))
+            {
+                uri = xmlNodeGetAttributeValue(node, "href");
+                break;
+            }
+        }
+        nodes->release();
+
+        if (uri.length() == 0)
+        {
+            Poco::XML::DOMWriter writer;
+            std::stringstream msg("Unable to process Elog subscription event from XML Element: ");
+            writer.writeNode(msg, &li_element);
+            BOOST_THROW_EXCEPTION(std::invalid_argument(msg.str()));
+        }
+
+        std::string const prefix = "/rw/elog/";
+        if (uri.find(prefix) != 0)
+            BOOST_THROW_EXCEPTION(std::invalid_argument{"Unable to process Elog subscription event: invalid resource URI \"" + uri + "\""});
+
+        std::vector<std::string> parts;
+        boost::split(parts, uri.substr(prefix.size()), boost::is_any_of("/"));
+
+        if (parts.size() < 2)
+            BOOST_THROW_EXCEPTION(std::invalid_argument{"Unable to process Elog subscription event: unable to find domain and sequence number in uri: \"" + uri + "\""});
+
+        ElogEvent event;
+        event.domain = std::stoi(parts[0]);
+        event.seqnum = std::stoi(parts[1]);
+        event.resource = std::make_shared<ElogSubscribableResource>(*this);
+        callback.processEvent(event);
+    }
+
+    std::vector<std::pair<int, std::string>> parseDomains(RWSResult const &rws_result)
+    {
+        std::vector<Poco::XML::Node *> domains_xml = xmlFindNodes(rws_result, XMLAttribute("class", "elog-domain-li"));
+        std::vector<std::pair<int, std::string>> elogDomains;
+        for (auto node : domains_xml)
+        {
+            std::pair<int, std::string> domain;
+            domain.first = std::stoi(xmlNodeGetAttributeValue(node, "title"));
+            domain.second = xmlFindTextContent(node, XMLAttribute("class", "domain-name"));
+            elogDomains.push_back(domain);
+        }
+        return elogDomains;
+    }
+
+    std::vector<std::pair<int, std::string>> getElogDomains(RWSClient &client, std::string lang)
+    {
+        std::stringstream uri;
+        uri << "/rw/elog?lang=" << lang << "&resource=count";
+        POCOResult poco_result = client.httpGet(uri.str());
+        RWSResult const &rws_result = parseXml(poco_result.content());
+        return parseDomains(rws_result);
+    }
+
+    ElogMessage parseElogMessageXml(std::string const xml_string)
+    {
+        Poco::AutoPtr<Poco::XML::Document> rws_result = parseXml(xml_string);
+        ElogMessage message;
+        message.messageType = static_cast<ElogMessageType>(std::stoi(xmlFindTextContent(rws_result, XMLAttribute("class", "msgtype"))));
```

Comment:
This `static_cast<ElogMessageType>` is slightly suspicious because it ignores potential invalid `enum` values.

Reply:
I added makeElogMessageType that throws exception on other values.